consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/blockcreation/BftBlockCreatorFactory.java

```diff
@@ -102,22 +105,35 @@ public Wei getMinTransactionGasPrice() {
   }
 
   public Bytes createExtraData(final int round, final BlockHeader parentHeader) {
-    final BftContext bftContext = protocolContext.getConsensusState(BftContext.class);
-    final ValidatorProvider validatorProvider = bftContext.getValidatorProvider();
-    checkState(validatorProvider.getVoteProvider().isPresent(), "Bft requires a vote provider");
-    final Optional<ValidatorVote> proposal =
-        validatorProvider.getVoteProvider().get().getVoteAfterBlock(parentHeader, localAddress);
-
-    final List<Address> validators =
-        new ArrayList<>(validatorProvider.getValidatorsAfterBlock(parentHeader));
-
-    final BftExtraData extraData =
-        new BftExtraData(
-            ConsensusHelpers.zeroLeftPad(vanityData, BftExtraDataCodec.EXTRA_VANITY_LENGTH),
-            Collections.emptyList(),
-            toVote(proposal),
-            round,
-            validators);
+    final BftExtraData extraData;
+    if (extraDataWithRoundInformationOnly) {
```

Comment:
Would prefer this wasn't part of the base BftBlockCreatorFactory and we had a specific QbftBlockCreatorFactory as it's specific to Qbft.

Reply:
refactored.